src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -73,6 +75,71 @@ public NewAddressModel GetNewAddress()
             return new NewAddressModel(base58Address);
         }
 
+        /// <summary>
+        /// RPC method that returns the spendable balance of all accounts.
+        /// Uses the first wallet and account.
+        /// </summary>
+        /// <returns>Total spendable balance of the wallet.</returns>
+        [ActionName("getbalance")]
+        [ActionDescription("Gets wallets spendable balance.")]
+        public Money GetBalance()
+        {
+            var account = this.GetAccount();
+
+            IEnumerable<AccountBalance> balances = this.walletManager.GetBalances(account.WalletName, account.AccountName);
+            return balances?.Sum(i => i.AmountConfirmed);
+        }
+
+        /// <summary>
+        /// RPC method to return transaction info from the wallet.
+        /// Uses the first wallet and account.
+        /// </summary>
+        /// <param name="txid">Transaction identifier to find.</param>
+        /// <returns>Transaction information.</returns>
+        [ActionName("gettransaction")]
+        [ActionDescription("Gets a transaction from the wallet.")]
+        public GetTransactionModel GetTransaction(string txid)
+        {
+            uint256 trxid;
+            if (!uint256.TryParse(txid, out trxid))
+                throw new ArgumentException(nameof(txid));
+
+            var accountReference = this.GetAccount();
+            var account = this.walletManager.GetAccounts(accountReference.WalletName)
+                                            .Where(i => i.Name.Equals(accountReference.AccountName))
+                                            .Single();
+
+            var transaction = account.GetTransactionsById(trxid)?.Single();
+
+            if (transaction == null)
+                return null;
+
+            var model = new GetTransactionModel
+            {
+                Amount = transaction.Amount,
+                BlockHash = transaction.BlockHash,
+                TransactionId = transaction.Id,
+                TransactionTime = transaction.CreationTime.ToUnixTimeSeconds(),
+                Details = new List<GetTransactionDetailsModel>(),
+                Hex = transaction.Hex
```

Comment:
I don't think it affects your code, but note that storing the hex is actually optional in our node, enabled via a CLI parameter. In bitcoind if there is missing data I think they return `""`. Not sure if hex is ever missing in a bitcoind wallet though.

Reply:
Yes I've seen that. I'll change it to return empty string if null.